### PR TITLE
Pause Engine in PMPro Core

### PR DESCRIPTION
**Pull request by dalemugford**
### All Submissions:

* [x] Have you followed the [Contributing guideline](https://github.com/strangerstudios/paid-memberships-pro/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/strangerstudios/paid-memberships-pro/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds a modular **Pause Engine** (`PMPro_Pause_Engine`) for progressive site lockdown during site migrations, or maintenance. This replaces the `PMPro_Migration_Site_Suspender` class in the PMPro Hosting plugin with a reusable, composable system any tooling can leverage.

The existing `pmpro_is_paused()` **remains untouched**, and only detects URL mismatches and halts Action Scheduler. It doesn't freeze membership changes, block gateway calls, suppress emails, or lock down the frontend. The hosting plugin's suspender handles some of this but wasn't reusable and blocked email permanently with no queue mechanism.

**Architecture:** A singleton orchestrator with 6 toggleable modules, activated in a defined order:

| Module | Slug | What it does |
|--------|------|-------------|
| Mutations | `pmpro_mutations` | Blocks checkout, level changes, and order creation |
| Gateways | `pmpro_gateways` | Blocks outbound HTTP to gateway APIs + returns 503 to inbound webhooks |
| Mail | `pmpro_mail` | Intercepts `wp_mail()`, queues to Action Scheduler for replay on resume |
| Schedules | `background_schedules` | Halts Action Scheduler and blocks cron spawning |
| Frontend | `frontend_block` | 503 maintenance page, blocks REST API, POST requests, and non-admin logins |
| Sessions | `logged_in_sessions` | Clears non-admin sessions on activation, blocks non-admin logins |

Two built-in presets: **Migration** (all 6 modules) and **Maintenance** (mutations + mail + schedules). Modules can be enabled/disabled individually at runtime for progressive lockdown. Admin bypass via `pmpro_manage_pause_mode` capability. Presets and modules are extensible via filters.

**Files added:**
- `classes/class-pmpro-pause-engine.php` — Interface, orchestrator, all 6 modules, convenience functions
- `tests/test-pause-engine.php` — Browser-based test harness (see testing instructions)

**Files modified:**
- `paid-memberships-pro.php` — Loads the pause engine on `plugins_loaded` priority 5
- `includes/admin.php` — Admin notice when engine is active with module list and "Resume All Services" button

**Public API:**
```php
PMPro_Pause_Engine::instance()->pause_with_preset( 'migration' );
PMPro_Pause_Engine::instance()->pause( array( 'pmpro_mutations', 'pmpro_gateways' ), 'my-tool' );
PMPro_Pause_Engine::instance()->enable_module( 'frontend_block' );
PMPro_Pause_Engine::instance()->disable_module( 'frontend_block' );
PMPro_Pause_Engine::instance()->resume();
pmpro_pause_engine_is_active();
pmpro_pause_module_is_active( 'pmpro_mutations' );
```

### How to test the changes in this Pull Request:

A browser-based test harness is included at `tests/test-pause-engine.php`. To set it up:

1. Copy the file into your site's `mu-plugins` directory: `cp wp-content/plugins/paid-memberships-pro/tests/test-pause-engine.php wp-content/mu-plugins/`
2. As an admin, navigate to `/wp-admin/?pmpro_pause=status` to open the test dashboard.
3. From there you can activate presets, enable custom module combinations, view live diagnostics, and resume — all via clickable links.

**Available URLs (append to `/wp-admin/`):**

| URL | Action |
|-----|--------|
| `?pmpro_pause=status` | Dashboard with module states, diagnostics, and quick action links |
| `?pmpro_pause=migration` | Activate the Migration preset (full lockdown) |
| `?pmpro_pause=maintenance` | Activate the Maintenance preset |
| `?pmpro_pause=custom&modules=pmpro_mutations,pmpro_gateways` | Activate specific modules |
| `?pmpro_pause=resume` | Resume all modules |
| `?pmpro_pause=test_mail` | Send a test email (verifies mail queuing) |

**What to verify:**

4. Activate the migration preset. Visit the site in an incognito window — you should see a 503 maintenance page.
5. Confirm your admin session remains fully functional while paused.
6. Check the status page diagnostics: gateway blocking, mail queue count, cron/AS halt status.
7. Click "Send a test email", return to status, confirm the queued email count increments.
8. Test webhook blocking: `curl -vk -X POST "https://yoursite.local/wp-admin/admin-ajax.php?action=stripe_webhook"` — should return 503.
9. Log out, log back in as admin — should succeed. Attempt non-admin login — should be rejected.
10. Hit resume, confirm all modules deactivate and the site returns to normal.
11. Remove the mu-plugin when done: `rm wp-content/mu-plugins/test-pause-engine.php`

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Added a modular Pause Engine (`PMPro_Pause_Engine`) for progressive site lockdown during migrations and maintenance.

### CODE REVIEW

File: tests/test-pause-engine.php [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+
+function pm_assert( $condition, $label ) {
+	global $pm_test_pass, $pm_test_fail;
+	if ( $condition ) {
+		$pm_test_pass++;
+		WP_CLI::success( $label );
+	} else {
+		$pm_test_fail++;
+		WP_CLI::error( $label, false );
+	}
```
[the remark this answers is not in the document]
Comment (dalemugford) [reply tests/test-pause-engine.php]: Stale.

File: tests/test-pause-engine.php [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+ * Quick pause engine test script.
+ *
+ * Run via WP-CLI:
+ *   wp eval-file wp-content/plugins/paid-memberships-pro/tests/test-pause-mode.php
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
```
[the remark this answers is not in the document]
Comment (dalemugford) [reply tests/test-pause-engine.php]: Stale.
